Code/Tools/AssetProcessor/native/utilities/PlatformConfiguration.cpp

```diff
@@ -1462,9 +1465,20 @@ namespace AssetProcessor
         {
             //using a bool instead of using #define UNIT_TEST because the user can also run batch processing in unittest
             m_scanFolders.push_back(source);
+
+            // since we're synthesizing folder adds, assign ascending folder ids if not provided.
+            if (source.ScanFolderID() == 0)
+            {
+                m_scanFolders.back().SetScanFolderID(m_scanFolders.size() - 1);
+            }
             return;
         }
 
+        AZ_Assert(
+            source.ScanPath() == AssetUtilities::NormalizeDirectoryPath(source.ScanPath()),
+            "AddScanFolder expects normalized name input but %s was given instead.",
```

Comment:
So this is condition comes from bad input? If so, why not just skip it and show an error? Is this really a coding error that requires a programmer to fix?

Reply:
I'm not quite sure what you would like to change here.   It is indeed a programming error (somewhere in AP) if we ever end up with a scan folder being added that isn't normalized, since all the places that call this function already normalize it beforehand, and thus this assert never triggers but future proofs it for if someone adds scan folders (in c++ code) by some other new means.

However, maybe I can change this so that the scan folder input isn't const, or I could copy it and normalize it here, instead of trying to assert, so it simply can't be wrong.  (This is not a bottleneck / hot path in the app)

Reply:
I'll just erase the assert.  It turns out all data members of ScanFolderInfo class are read-only (ie, there are no setters) and the constructor normalizes the path from whatever was passed in, so its impossible for this assert to trigger.